include/trieste/passes.h

```diff
@@ -112,65 +113,178 @@ namespace trieste
     std::chrono::microseconds duration;
   };
 
+  struct ProcessResult
+  {
+    bool ok;
+    std::string last_pass;
+    Node ast;
+    Node error;
```

Comment:
Would this be better as a `Nodes`?  Then you don't need to introduce an `ErrorSeq` token?